TypeScript Files/src/chooseChioce.ts

```diff
@@ -0,0 +1,132 @@
+import _orderBy from 'lodash/orderBy';
+import getArrayObject from './getArrayObject';
+import displayfun from './displayfunc';
+import { MyStrings, Iobjectarray , IUnionstringundefined , IUnionRequiredarrayundefined } from './enum.d';
+
+let chooseChioce = (finalArrayObject : Required<Iobjectarray[]>) => {
+    const choice = require('prompt-sync')();
+    const userarray = require('prompt-sync')();
+    
+    /////////// CHOOSE LIST //////////////////////
+    let chooseoption = (y : string ='first') : IUnionstringundefined => {
+        console.log("A : for add order. \n");
+        console.log("B : for delet order. \n");
+        console.log("C : find an order. \n");
+        console.log("D : for checkout. \n");
+        console.log("Q : for Exit. \n");
+        let option : IUnionstringundefined;
+        if(y === 'first'){
+            option = choice('Please enter your choice : ');
+        }
+        else if( y=== 'another choice'){
+            option = choice('Please enter another choice or Q for Exit: ');
+        }
+        else{
+            option = choice('Please enter a correct choice from above (make sure your using UperCase): ');
+        }
+        return option;
+    }
+    
+    /////////// ADD ORDER //////////////////////
+    let addorder = () : IUnionRequiredarrayundefined => {
+        console.log("NOTE : charecters, hexadecimals + float numbers will be ignored, only the positive integer values are allowed ");
+        let arraybyuser = userarray("Enter your value/values using ',' in between (ex: for one value 43/ for values : 43,23,123): ");
+        arraybyuser = arraybyuser
+                    .split(',')
+                    .filter(element => {
+                        if (!isNaN(element) && element.toString().indexOf('.') == -1 && element.toString().indexOf('#') == -1){
+                            return element;
+                        }
+                    }).map(element => parseInt(element))
+                    .filter(element => Number.isInteger(parseInt(element)) && element > 0)
+                    .sort((a,b) => a-b);
+        if(arraybyuser.length > 0){
+            arraybyuser.map((element) => {
+                finalArrayObject.push(getArrayObject(element));
+            });
+            console.log('the total Order numbers is/are : ');
+            console.log(_orderBy(finalArrayObject,['totalOrderPrice'],['asc']).map(e=>{return e=e.totalNumber}));
+            finalArrayObject = _orderBy(finalArrayObject,['totalOrderPrice'],['asc']);
+            return finalArrayObject;
+        }
+        else{
+            console.log("the value you entered is not correct, please check the note below : ");
+            addorder();
+        }
+    };
+    
+    /////////// DELETE ORDER //////////////////////
+    let deleteorder = () : IUnionRequiredarrayundefined => {
```

Comment:
let's put all these functions separately, this file is a bit too big.

Reply:
done